src/configs/all.ts

```diff
@@ -0,0 +1,255 @@
+    // "switch-default": Won't this be deprecated?
+    "triple-equals": true,
+    // "typeof-compare": Won't this be deprecated?
+    "use-isnan": true,
+
+    // Maintainability
+
+    "cyclomatic-complexity": true,
+    "eofline": true,
+    "indent": [true, "spaces"],
+    "linebreak-style": [true, "LF"],
+    "max-classes-per-file": [true, 1],
+    "max-file-line-count": [true, 1000],
+    "max-line-length": [true, 120],
+    "no-default-export": true,
+    "no-mergeable-namespace": true,
+    "no-require-imports": true,
+    "no-trailing-whitespace": true,
+    "object-literal-sort-keys": true,
+    "prefer-const": true,
+    "trailing-comma": [true, {
+        "multiline": "always",
+        "singleline": "never",
+    }],
+
+    // Style
+
+    "align": [true,
+        "parameters",
+        "arguments",
+        "statements",
+    ],
+    "array-type": [true, "array-simple"],
+    "arrow-parens": true,
+    "arrow-return-shorthand": [true, "multiline"],
+    "callable-types": true,
+    "class-name": true,
+    "comment-format": [
+        true,
+        "check-space",
+        "check-uppercase",
+    ],
+    "completed-docs": true,
+    // "file-header": No sensible default
+    "import-spacing": true,
+    "interface-name": true,
+    "interface-over-type-literal": true,
+    "jsdoc-format": true,
+    "match-default-export-name": true,
+    "new-parens": true,
+    "newline-before-return": true,
+    "no-angle-bracket-type-assertion": true,
+    "no-boolean-literal-compare": true,
+    "no-consecutive-blank-lines": true,
+    "no-parameter-properties": true,
+    "no-reference-import": true,
+    "no-unnecessary-callback-wrapper": true,
+    "no-unnecessary-initializer": true,
+    "no-unnecessary-qualifier": true,
+    "object-literal-key-quotes": [true, "consistent-as-needed"],
+    "object-literal-shorthand": true,
+    "one-line": [true,
+        "check-catch",
+        "check-else",
+        "check-finally",
+        "check-open-brace",
+        "check-whitespace",
+    ],
+    "one-variable-per-declaration": true,
+
+    "ordered-imports": [true, {
+        "import-sources-order": "case-insensitive",
+        "named-imports-order": "case-insensitive",
+    }],
+    "prefer-function-over-method": true,
+    "prefer-method-signature": true,
+    "quotemark": [true, "double", "avoid-escape"],
+    "semicolon": [true, "always"],
+    "space-before-function-paren": [true, {
+        "anonymous": "never",
+        "asyncArrow": "always",
+        "constructor": "never",
+        "method": "never",
+        "named": "never",
+    }],
+    "variable-name": [
+        true,
+        "ban-keywords",
+        "check-format",
+    ],
+    "whitespace": [
+        true,
+        "check-branch",
+        "check-decl",
+        "check-operator",
+        "check-module",
+        "check-separator",
+        "check-type",
+        "check-typecast",
+        "check-preblock",
+    ],
+};
+
+// Exclude typescript-only rules from jsRules
```

Comment:
we should be able to infer this from rule metadata. if the metadata is wrong, let's fix that